libs/core/src/main/java/org/opensearch/common/io/stream/Writeable.java

```diff
@@ -5,73 +5,101 @@
  * across the wire" using OpenSearch's internal protocol. If the implementer also implements equals and hashCode then a copy made by
  * serializing and deserializing must be equal and have the same hashCode. It isn't required that such a copy be entirely unchanged.
  *
  * @opensearch.internal
  */
-public interface BaseWriteable<S extends BaseStreamOutput> {
+public interface Writeable {
     /**
      * A WriteableRegistry registers {@link Writer} methods for writing data types over a
-     * {@link BaseStreamOutput} channel and {@link Reader} methods for reading data from a
-     * {@link BaseStreamInput} channel.
+     * {@link StreamOutput} channel and {@link Reader} methods for reading data from a
+     * {@link StreamInput} channel.
      *
      * @opensearch.internal
      */
     class WriteableRegistry {
-        private static final Map<Class<?>, Writer<? extends BaseStreamOutput, ?>> WRITER_REGISTRY = new ConcurrentHashMap<>();
+        private static final Map<Class<?>, Writer<?>> WRITER_REGISTRY = new ConcurrentHashMap<>();
         private static final Map<Class<?>, Class<?>> WRITER_CUSTOM_CLASS_MAP = new ConcurrentHashMap<>();
-        private static final Map<Byte, Reader<? extends BaseStreamInput, ?>> READER_REGISTRY = new ConcurrentHashMap<>();
+        private static final Map<Byte, Reader<?>> READER_REGISTRY = new ConcurrentHashMap<>();
 
         /**
          * registers a streamable writer
          *
          * @opensearch.internal
          */
-        public static <W extends Writer<? extends BaseStreamOutput, ?>> void registerWriter(final Class<?> clazz, final W writer) {
+        public static <W extends Writer<?>> void registerWriter(final Class<?> clazz, final W writer) {
             if (WRITER_REGISTRY.putIfAbsent(clazz, writer) != null) {
                 throw new IllegalArgumentException("Streamable writer already registered for type [" + clazz.getName() + "]");
             }
+            WRITER_REGISTRY.put(clazz, writer);
         }
 
         /**
          * registers a streamable reader
          *
          * @opensearch.internal
          */
-        public static <R extends Reader<? extends BaseStreamInput, ?>> void registerReader(final byte ordinal, final R reader) {
+        public static <R extends Reader<?>> void registerReader(final byte ordinal, final R reader) {
             if (READER_REGISTRY.putIfAbsent(ordinal, reader) != null) {
                 throw new IllegalArgumentException("Streamable reader already registered for ordinal [" + (int) ordinal + "]");
             }
+            READER_REGISTRY.put(ordinal, reader);
         }
 
         public static void registerClassAlias(final Class<?> classInstance, final Class<?> classGeneric) {
-            if (WRITER_CUSTOM_CLASS_MAP.putIfAbsent(classInstance, classGeneric) != null) {
+            if (WRITER_CUSTOM_CLASS_MAP.containsKey(classInstance)) {
```

Comment:
Think we should use `putIfAbsent` here to be consistent in this class.

Reply:
Agree!